src/main/java/co/cask/bigquery/BigQuerySource.java

```diff
@@ -237,6 +237,7 @@ public Schema getSchema(Request request) throws Exception {
       for (Field field : bgSchema.getFields()) {
         Field.Type type = field.getType();
         Schema.Type cType = Schema.Type.STRING;
+        Schema.LogicalType lType = null;
```

Comment:
cleaner to just have a Schema here instead of cType and lType. In each clause, set a Schema.